SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -1787,7 +1791,26 @@ void USpatialNetDriver::ServerReplicateActors_ProcessPrioritizedActors(UNetConne
 
 #endif // WITH_SERVER_CODE
 
-thread_local TArray<AActor*> GSenderStack;
+struct SenderActorDesc
+{
+	enum ItemKind
+	{
+		Sender,
+		Dependent,
+		Resolution
+	};
+
+	SenderActorDesc(AActor* InActor, ItemKind InKind)
+		: Actor(InActor)
+		, Kind(InKind)
+	{
+	}
+	AActor* const Actor;
+	ItemKind Kind;
+	bool bHasBeenUsed = false;
+};
+
+thread_local TArray<SenderActorDesc> GSenderStack;
```

Comment:
I know this was in the last pr, but why is this thread_local? I can't really find any reason not to just have it as a member.

Reply:
Discussion continuing offline

Reply:
Interested in a summary.

Reply:
Ditto - i was going to comment but it was in a previous submit... also it is not in a private namespace.

Reply:
So using a thread_local is a way to pass "hidden" arguments on the stack.
The situation is that I need an additional Sender Actor to be passed alongside RPC calls. 
In order to keep the impact on the engine minimal, I did not add additional parameters to all the involved function. Instead, I rely on NetDriver calls to push the arguments in the thread_local stack, which is used later when we receive the ProcessRPC call.
Using a member instead could achieve the same results, but 
1) In terms of object design, it does not have the same meaning as other members. I find it better to keep transient state separated from persistent state
2) It would fall apart in multithread (ok we are not multithreaded, but doing it this way every time you have to means it will never be an issue).

Reply:
I see the benefit now, cool. Can we maybe leave a comment in the code detailing this? I'm afraid this is far from a common pattern.